docs/hub/concepts/response_formatting_and_structured_data.md

```diff
@@ -0,0 +1,207 @@
+
+# Response Formatting and Structure
+
+Guardrails provides several interfaces to help llms respond in valid JSON which can then be validated using Guardrails validators.
+
+## Prerequisites
+
+First, create a Pydantic model and guard for the structured data. The model should describe the data structure to be returned. Fields do support  We"ve included here a sample prompt and data to extract the structured data from. 
+
+```py
+from guardrails import Guard
+from guardrails.hub import RegexMatch
+from pydantic import BaseModel, Field
+from typing import List
+
+NAME_REGEX = "^[A-Z][a-z]+\s[A-Z][a-z]+$"
+
+class Delivery(BaseModel):
+    custome_name: str=Field(validators=[RegexMatch(regex=NAME_REGEX)], description="customer name")
+    pickup_time: str=Field(description="date and time of pickup")
+    pickup_location: str=Field(description="address of pickup")
+    dropoff_time: str=Field(description="date and time of dropoff")
+    dropoff_location: str=Field(description="address of dropoff")
+    price: str = Field(description="price of delivery with currency symbol included")
+
+class Schedule(BaseModel):
+    deliveries: List[Delivery]
+
+guard = Guard.from_pydantic(Schedule)
+chat_history="""
+nelson and murdock: i need a pickup 797 9th Avenue, manila envelope, June 3 10:00am with dropoff 10:30am Courthouse, 61 Center Street C/O frank james
+operator: quote - $23.00
+neslon and murdock: perfect, we accept the quote
+operator: 797 9th ave, 10:00am pickup comfirmed
+abc flowers: i need a pickup of a flowers from abc flowers at 21 3rd street at 11:00am on june 2 with a dropoff at 75th Ave at 5:30pm same day
+operator: 21 3rd street flowers quote - $14.50
+abc flowers: accepted
+polk and wardell: i need a pickup of a bagels from Bakers Co at 331 5th street at 11:00am on june 3 with a dropoff at 75th Ave at 5:30pm same day
+operator: 331 5th street bagels quote - $34.50
+polk and wardell: accepted
+"""
+
+prompt = """
+From the chat exchanges below extract a schedule of deliveries.
+Chats:
+${chat_history}
+"""
+```
+
+## Optional Guard.__call__ llm_api callable
+
+`guard(` calls will now default to liteLLM"s callable and OpenAI compatible interface if model is set and no llm_api is provided
+
+```py
+from guardrails import Guard
+from guardrails.hub import RegexMatch
+
+NAME_REGEX = "^[A-Z][a-z]+\s[A-Z][a-z]+$"
+guard = Guard()
+
+response = guard(
+    model="gpt-4o",
+    instructions="You are a helpful assistant.",
+    prompt="How many moons does jupiter have?",
+)
+```
+
+## Function/tool calling structured response formatting
```

Comment:
Since the above section is `Prerequisites`, I think a high level heading could make sense here before listing out the different options as sub sections beneath it.

```md
## Some heading that makes sense

### Function/tool calling structured response formatting
```